### ISSUE TITLE
[Heartbeat] root privileges required for ICMP

### ISSUE THREAD

**Issue by fkelbert** [elastic/beats#11498]
It seems root privileges are required to execute the following monitor:

```
- type: icmp
  hosts: [ "google.co.uk", "8.8.8.8" ]
  schedule: '@every 10s'
```

When running without root privileges, heartbeat fails with the following message: `Exiting: could not create monitor: job err failed to initiate IPv4 support`

I propose to:
- include a hint towards missing root privileges
- extend the documentation accordingly

**Comment by andrewvc** [comment]
I agree, the error message is terrible. We should work on it.

Even more ideally, we could research using capabilities instead of root access to let people run with the minimum of necessary privileges.

**Comment by leopucci** [comment]
This pull request refers to https://github.com/elastic/beats/issues/11498

I've added log messages and unprivileged icmp access.

**- Non-root execution**
Setting CAP_NET_RAW on file
Can be set by executing setcap cap_net_raw+ep heartbeat to make raw socket available

**- Unprivileged icmp access** 
Available for Linux/Mac OS darwin:
sysctl -w net.ipv4.ping_group_range="0 100"
After allowing group ids, users can use unpriveleged icmp socket

**Comment by andrewvc** [comment]
Fixed with #13795
